### PR TITLE
Fix integration tests

### PR DESCRIPTION
**Pull request by nevalla**
This PR fixes integration test failures.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,9 +19,9 @@
     "compile:main": "webpack --config webpack.main.ts",
     "compile:renderer": "webpack --config webpack.renderer.ts",
     "compile:i18n": "lingui compile",
-    "build:linux": "yarn compile && electron-builder --linux --dir -c.productName=LensDev",
-    "build:mac": "yarn compile && electron-builder --mac --dir -c.productName=LensDev",
-    "build:win": "yarn compile && electron-builder --win --dir -c.productName=LensDev",
+    "build:linux": "yarn compile && electron-builder --linux --dir -c.productName=Lens",
```
Comment (nevalla) [inline package.json:22]: What was the benefit of using `LensDev` as `productName`?

File: src/main/index.ts [hunk 2]
```diff
@@ -21,7 +21,9 @@ import logger from "./logger"
 
 const workingDir = path.join(app.getPath("appData"), appName);
 app.setName(appName);
-app.setPath("userData", workingDir);
+if(!process.env.CICD) {
+  app.setPath("userData", workingDir);
```
Comment (nevalla) [inline src/main/index.ts:25]: We need to skip this to have unique userData between test cases

File: src/main/window-manager.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export class WindowManager {
         nodeIntegration: true,
         enableRemoteModule: true,
         webviewTag: true,
+        nodeIntegrationInSubFrames: true
```
Comment (nevalla) [inline src/main/window-manager.ts]: Oops, some leftover. Will remove this.

File: src/main/window-manager.ts [hunk 4]
```diff
@@ -73,7 +74,7 @@ export class WindowManager {
       await this.showSplash();
       await this.mainView.loadURL(`http://localhost:${this.proxyPort}`)
       this.mainView.show();
-      this.splashWindow.hide();
+      this.splashWindow.close();
```
Comment (nevalla) [inline src/main/window-manager.ts:76]: Hiding splashWindow leaves window in background causing troubles in integration tests where windowCount is calculated.

File: src/main/window-manager.ts [hunk 5]
```diff
@@ -89,6 +90,9 @@ export class WindowManager {
         frame: false,
         resizable: false,
         show: false,
+        webPreferences: {
+          nodeIntegration: true
```
Comment (nevalla) [inline src/main/window-manager.ts:93]: Integration tests requires nodeIntegration, and it is determined by window's nodeIntegration option, so we need to enable it also for splashWindow.

File: src/renderer/components/+add-cluster/add-cluster.tsx [hunk 6]
```diff
@@ -71,8 +71,9 @@ export class AddCluster extends React.Component {
     if (value instanceof KubeConfig) {
       const context = value.currentContext;
       const isNew = userStore.newContexts.has(context);
+      const className = `${context} kube-context flex gaps align-center`
       return (
-        <div className="kube-context flex gaps align-center">
+        <div className={className}>
```
Comment (nevalla) [inline src/renderer/components/+add-cluster/add-cluster.tsx:76]: We can select correct kube context in integration test by adding kubecontext name to `className` property.

File: integration/helpers/utils.ts [hunk 7]
```diff
@@ -3,31 +3,35 @@ import { Application } from "spectron";
...
 
 export function setup(): Application {
+  const userData = process.cwd() + "/.spectron"
   return new Application({
     // path to electron app
     args: [],
     path: appPath,
     startTimeout: 30000,
     waitTimeout: 30000,
+    chromeDriverArgs: ['remote-debugging-port=9222'],
```
Comment (nevalla) [inline integration/helpers/utils.ts:23]: https://github.com/electron-userland/spectron/issues/353#issuecomment-522846725

File: package.json (hunk 1 above)
Comment (jakolehm) [reply package.json:22]: I think it originates from electron-webpack (not used anymore) where we did have `Lens` for production and `Electron` as dev/test.

File: package.json [hunk 8]
```diff
@@ -33,7 +33,7 @@
     "download:kubectl": "yarn run ts-node build/download_kubectl.ts",
     "download:helm": "yarn run ts-node build/download_helm.ts",
     "lint": "eslint $@ --ext js,ts,tsx --max-warnings=0  src/",
-    "rebuild-pty": "electron-rebuild -f -w node-pty"
+    "rebuild-pty": "yarn run electron-rebuild -f -w node-pty"
```
Comment (jakolehm) [inline package.json:36]: Just a note that I don't know why we even need `rebuild-pty`.

**Comment by Nokel81** [comment]
Looks good to me

**Review by jakolehm — APPROVED** [review]

**Pull request merged**